### PR TITLE
fix(editor): Connecting nodes to triggers when adding them together

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
This PR fixes a bug where node is not connected to trigger node when they are added together from the node creator panel.

## Related tickets and issues
Fixes ADO-1933

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts [hunk 1]
```diff
@@ -163,6 +163,20 @@ export const useActions = () => {
...
 
+	/**
+	 * Checks if added nodes contain trigger followed by another node
+	 * In this case, we should connect the trigger with the following node
+	 */
+	function shouldConnectWithExistingTrigger(addedNodes: AddedNode[]): boolean {
+		const { selectedView } = useNodeCreatorStore();
+		if (selectedView === TRIGGER_NODE_CREATOR_VIEW && addedNodes.length === 2) {
+			const isTriggerNode = useNodeTypesStore().isTriggerNode(addedNodes[0].type);
+			console.log('isTriggerNode', isTriggerNode, addedNodes[0].type);
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts:174]: 👁️

File: packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts [hunk 2]
```diff
@@ -163,6 +163,20 @@ export const useActions = () => {
 		};
 	}
 
+	/**
+	 * Checks if added nodes contain trigger followed by another node
+	 * In this case, we should connect the trigger with the following node
+	 */
+	function shouldConnectWithExistingTrigger(addedNodes: AddedNode[]): boolean {
+		const { selectedView } = useNodeCreatorStore();
+		if (selectedView === TRIGGER_NODE_CREATOR_VIEW && addedNodes.length === 2) {
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts:172]: Do we need to check for the `TRIGGER_NODE_CREATOR_VIEW` here? What if user is at the regular view but added the node by searching for it(which would show nodes from both views)?

Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts:172]: In that case we don't need it. I wasn't sure about all of the options to add nodes but decided to go conservative with the fix. Will remove

**Review by OlegIvaniv — DISMISSED** [review]
Looking good

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.37.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.37.0)

**Pull request merged**